test/benchmarks/randomized_benchmarking.py

```diff
@@ -15,40 +15,164 @@
 # pylint: disable=import-error
 
 import os
+from functools import lru_cache
+
 import numpy as np
-from qiskit_experiments.library import StandardRB
 
-try:
-    from qiskit.compiler import transpile
+from qiskit.compiler import transpile
+from qiskit.circuit import QuantumCircuit
+from qiskit.quantum_info import random_clifford
+from qiskit.circuit import Gate
+
+
+class VGate(Gate):
+    """V Gate used in Clifford synthesis."""
+
+    def __init__(self):
+        """Create new V Gate."""
+        super().__init__("v", 1, [])
+
+    def _define(self):
+        """V Gate definition."""
+        qc = QuantumCircuit(1)
+        qc.sdg(0)
+        qc.h(0)
+        self.definition = qc
```

Comment:
I think so, the only question is whether you think there is value in doing a backfill run from before `SingletonGate` was added. If you think we'll want to run this benchmark on historical commits then let's leave it otherwise lets update it.

Reply:
Particularly given the circuit under consideration here, it would be cool to see the effects before and after singleton gates, but we can't actually do that with a single trace, I think?

It's getting tricky to think about the traces, because I think so many of our traces just stop at arbitrary points, or the tests are written in such old Terra style that they're barely relevant anymore.

Reply:
We should be able to, I can trigger a specific back run of this benchmark from before singletons untill after. I think the code here should be compatible with either form.

Reply:
It's more that before singletons are added to the library, `VGate` obviously can't be defined as one, but after, we can't change `VGate` to be one without asv considering it a new trace, I think, because we'd be changing the setup code?  Unless it won't notice that we changed code from outside the actual test / setup functions.  I can't remember how much it hashes.

Reply:
Oh, I meant more we just leave `VGate` and `WGate` as is we could do a backfill run. If they inherit from `Gate` then the code here should work before singletons and after singletons were introduced. We'd still see a noticeable speedup from singletons if we did a backfill run because all the other gates become singletons. It wouldn't be a 100% singleton circuit anymore but it would mostly be. That's why the choice is between if we want to do a backfill run we leave this as-is, if we don't care then I'll convert it to a singleton gate.

The other option is I can just remove the custom gates and inline their definitions where they're used.

Reply:
Yeah, let's just leave them as-is and have a not-100%-singleton run.  It's not like our benchmarks are fully up-to-date with best Terra practices across the board.